extension/js/addTopicButtons.js

```diff
@@ -0,0 +1,265 @@
+    if(usernameE){
+       username = usernameE.innerText
+    }
+
+    var idE = document.querySelector(`#${prefix}${post_id} .username`);
+    if (!idE){
+       var usernameE = document.querySelector(`#p${prefix}${post_id} .username-coloured`)
+    }
+    var usernameLink = ""
+    let id =""
+    if(idE){
+       usernameLink = idE.href;
+       id = usernameLink.split("u=")[1];
+    }
+    var tsE = document.querySelector(`#${prefix}${post_id} time`)
+    if (!tsE){
+        var tsE = document.querySelector(`#${prefix}${post_id} [href='#postingbox']`).getAttribute('onclick')
+        if(tsE)
+            var time = tsE.match('(?<=time:)(.*)(?=,user)')[0];
+    }else{
+        var ts = Date.parse(tsE.dateTime)
+        var time = ts / 1000
+    }
+
+
+
+    return {
+        "username":username,
+        "id":id,
+        "time":time
+    };
+}
+
+function ping_user(post_id){
+
+    if (window.location.href.includes("posting.php")){
+        let PostDetails = getPostDetails(post_id,"pr")
+        let text = `[quote="${PostDetails.username}" user_id=${PostDetails.id} time=${PostDetails.time} post_id=${post_id}]\n[/quote]`
+        insert_text(text)
+    }else{
+        let PostDetails = getPostDetails(post_id)
+        let text = `[quote="${PostDetails.username}" user_id=${PostDetails.id} time=${PostDetails.time} post_id=${post_id}]\n[/quote]`
+
+        try {
+            addText(text)
+        }catch (exception_var) {
+			  navigator.clipboard.writeText(text)
+        }
+    }
+}
+
+function addText(text){
+    var textarea = document.querySelector("#message-box textarea");
+
+    if (!isNaN(textarea.selectionStart)) {
+    	var sel_start = textarea.selectionStart;
+    	var sel_end = textarea.selectionEnd;
+    	mozWrapApp(textarea, text, '');
+    	textarea.selectionStart = sel_start + text.length;
+    	textarea.selectionEnd = sel_end + text.length;
+    } else if (textarea.createTextRange && textarea.caretPos) {
+    	if (baseHeight !== textarea.caretPos.boundingHeight) {
+    		textarea.focus();
+    		storeCaret(textarea);
+    	}
+    	var caret_pos = textarea.caretPos;
+    	caret_pos.text = caret_pos.text.charAt(caret_pos.text.length - 1) === ' ' ? caret_pos.text + text + ' ' : caret_pos.text + text;
+    } else {
+    	textarea.value = textarea.value + text;
+    }
+    textarea.focus();
+}
+function mozWrapApp(txtarea, open, close) {
+	var selLength = (typeof(txtarea.textLength) === 'undefined') ? txtarea.value.length : txtarea.textLength;
+	var selStart = txtarea.selectionStart;
+	var selEnd = txtarea.selectionEnd;
+	var scrollTop = txtarea.scrollTop;
+
+	var s1 = (txtarea.value).substring(0,selStart);
+	var s2 = (txtarea.value).substring(selStart, selEnd);
+	var s3 = (txtarea.value).substring(selEnd, selLength);
+
+	txtarea.value = s1 + open + s2 + close + s3;
+	txtarea.selectionStart = selStart + open.length;
+	txtarea.selectionEnd = selEnd + open.length;
+	txtarea.focus();
+	txtarea.scrollTop = scrollTop;
+
+	return;
```

Comment:
```suggestion
```
Warning:(254, 2) 'return' is unnecessary as the last statement in a function with no return value